src/transformers/models/resnet/modeling_tf_resnet.py

```diff
@@ -0,0 +1,491 @@
+        hidden_state = self.convolution(hidden_state)
+        hidden_state = self.normalization(hidden_state, training=training)
+        hidden_state = self.activation(hidden_state)
+        return hidden_state
+
+
+class TFResNetEmbeddings(tf.keras.layers.Layer):
+    """
+    ResNet Embeddings (stem) composed of a single aggressive convolution.
+    """
+
+    def __init__(self, config: ResNetConfig, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.embedder = TFResNetConvLayer(
+            config.embedding_size,
+            kernel_size=7,
+            stride=2,
+            activation=config.hidden_act,
+            name="embedder",
+        )
+        self.pooler = tf.keras.layers.MaxPool2D(pool_size=3, strides=2, padding="valid", name="pooler")
+        self.num_channels = config.num_channels
+
+    def call(self, pixel_values: tf.Tensor, training: bool = False) -> tf.Tensor:
+        _, num_channels, _, _ = shape_list(pixel_values)
+        if tf.executing_eagerly() and num_channels != self.num_channels:
+            raise ValueError(
+                "Make sure that the channel dimension of the pixel values match with the one set in the configuration."
+            )
+        hidden_state = pixel_values
+        hidden_state = self.embedder(hidden_state)
+        hidden_state = tf.pad(hidden_state, [[0, 0], [0, 0], [1, 1], [1, 1]])
+        # (batch_size, num_channels, height, width) -> (batch_size, height, width, num_channels)
+        hidden_state = tf.transpose(hidden_state, (0, 2, 3, 1))
+        hidden_state = self.pooler(hidden_state)
+        # (batch_size, height, width, num_channels) -> (batch_size, num_channels, height, width)
+        hidden_state = tf.transpose(hidden_state, (0, 3, 1, 2))
+        return hidden_state
+
+
+class TFResNetShortCut(tf.keras.layers.Layer):
+    """
+    ResNet shortcut, used to project the residual features to the correct size. If needed, it is also used to
+    downsample the input using `stride=2`.
+    """
+
+    def __init__(self, out_channels: int, stride: int = 2, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.convolution = tf.keras.layers.Conv2D(
+            out_channels, kernel_size=1, strides=stride, use_bias=False, name="convolution"
+        )
+        # Use same default momentum and epsilon as PyTorch equivalent
+        self.normalization = tf.keras.layers.BatchNormalization(
+            axis=1, epsilon=1e-5, momentum=0.1, name="normalization"
+        )
+
+    def call(self, x: tf.Tensor, training: bool = False) -> tf.Tensor:
+        hidden_state = x
+        # (batch_size, num_channels, height, width) -> (batch_size, height, width, num_channels)
+        hidden_state = tf.transpose(hidden_state, (0, 2, 3, 1))
+        hidden_state = self.convolution(hidden_state)
+        # (batch_size, height, width, num_channels) -> (batch_size, num_channels, height, width)
+        hidden_state = tf.transpose(hidden_state, (0, 3, 1, 2))
```

Comment:
Same here, could this be an NCHW convolution instead?

Reply:
The reason I have it this way is that Conv2D can't run on CPU with `channels_first `. You get this error: 

```
The Conv2D op currently only supports the NHWC tensor format on the CPU. The op was given the format: NCHW [Op:Conv2D]```